airbyte-integrations/connectors/source-e2e-test/src/main/java/io/airbyte/integrations/source/e2e_test/ContinuousFeedConfig.java

```diff
@@ -0,0 +1,203 @@
+    final MockCatalogType mockCatalogType = MockCatalogType.valueOf(mockCatalogConfig.get("type").asText());
+    switch (mockCatalogType) {
+      case SINGLE_STREAM -> {
+        final String streamName = mockCatalogConfig.get("stream_name").asText();
+        final String streamSchemaText = mockCatalogConfig.get("stream_schema").asText();
+        final Optional<JsonNode> streamSchema = Jsons.tryDeserialize(streamSchemaText);
+        if (streamSchema.isEmpty()) {
+          throw new JsonValidationException(String.format("Stream \"%s\" has invalid schema: %s", streamName, streamSchemaText));
+        }
+        processSchema(streamSchema.get());
+        checkSchema(streamName, streamSchema.get());
+
+        final AirbyteStream stream = new AirbyteStream().withName(streamName).withJsonSchema(streamSchema.get());
+        return new AirbyteCatalog().withStreams(Collections.singletonList(stream));
+      }
+      case MULTI_STREAM -> {
+        final String streamSchemasText = mockCatalogConfig.get("stream_schemas").asText();
+        final Optional<JsonNode> streamSchemas = Jsons.tryDeserialize(streamSchemasText);
+        if (streamSchemas.isEmpty()) {
+          throw new JsonValidationException("Input stream schemas are invalid: %s" + streamSchemasText);
+        }
+
+        final List<AirbyteStream> streams = new LinkedList<>();
+        for (final Map.Entry<String, JsonNode> entry : MoreIterators.toList(streamSchemas.get().fields())) {
+          final String streamName = entry.getKey();
+          final JsonNode streamSchema = Jsons.clone(entry.getValue());
+          processSchema(streamSchema);
+          checkSchema(streamName, streamSchema);
+          streams.add(new AirbyteStream().withName(streamName).withJsonSchema(streamSchema));
+        }
+        return new AirbyteCatalog().withStreams(streams);
+      }
+      default -> throw new IllegalArgumentException("Unsupported mock catalog type: " + mockCatalogType);
+    }
+  }
+
+  /**
+   * Validate the stream schema against Json schema draft 07.
+   */
+  private static void checkSchema(final String streamName, final JsonNode streamSchema) throws JsonValidationException {
+    final Set<String> validationMessages = SCHEMA_VALIDATOR.validate(JSON_SCHEMA_DRAFT_07, streamSchema);
+    if (!validationMessages.isEmpty()) {
+      throw new JsonValidationException(String.format(
+          "Stream \"%s\" has invalid schema.\n- Errors: %s\n- Schema: %s",
+          streamName,
+          Strings.join(validationMessages, "; "),
+          streamSchema.toString()));
+    }
+  }
+
+  /**
+   * Patch the schema so that 1) it allows no additional properties, and 2) all fields are required.
+   * This is necessary because the mock Json object generation library may add extra properties, or
+   * omit non-required fields. TODO (liren): patch the library so we don't need to patch the schema
+   * here.
+   */
+  private static void processSchema(final JsonNode schema) {
```

Comment:
Yes, these two behaviors are correct.

However, currently we cannot precisely customize the random Json object generator on 1) how many additional properties to add, and 2) how many non-required fields to include. 

Regarding additional properties, the library may generate a field with horrible names, like: `adf23rj2flfv032jr;af;d`, or even non-ascii characters. It makes the object looks ugly. So I turned off the additional property generation for now.

Regarding non-required fields, though there is a `nonRequiredPropertyChance` setting in the config class, it does not seem to work very well. Even if the change is currently set to 1.0, the generated object can still be empty, if no field is required. To avoid emitting lots of objects, I marked all fields as required.

I will add more comments in the Javadoc to provide the context.